devlab/tests/functional_test.py

```diff
@@ -31,3 +34,97 @@ class FunctionalTest(unittest.TestCase):
+        subnets = [cfg.extend(i['subnets'])
+                   for i in config.tenants if 'subnets' in i]
+        return self.__get_neutron_resources('subnets', subnets)
+
+    def filter_routers(self):
+        routers = [i['router']['name'] for i in config.routers]
+        return self.__get_neutron_resources('routers', routers)
+
+    def filter_floatingips(self):
+        # Now we create floating ip, after tenant networks created.
+        # Will be fixed with tests for floating ip associating
+        def get_fips(_user):
+            self.src_cloud.switch_user(user=_user['name'],
+                                       tenant=_user['tenant'],
+                                       password=_user['password'])
+            _client = self.src_cloud.neutronclient
+            return [_fip['floating_ip_address']
+                    for _fip in _client.list_floatingips()['floatingips']]
+
+        for tenant in config.tenants:
+            fips = [fip for user in config.users
+                    if tenant['name'] == user['tenant'] and user['enabled']
+                    and not user.get('deleted')
+                    for fip in get_fips(user)]
+            return set(fips)
+
+    def filter_users(self):
+        users = [i['name'] for i in config.users]
+        return self.__get_keystone_resources('users', users)
+
+    def filter_tenants(self):
+        tenants = [i['name'] for i in config.tenants]
+        return self.__get_keystone_resources('tenants', tenants)
+
+    def filter_roles(self):
+        roles = [i['name'] for i in config.roles]
+        return self.__get_keystone_resources('roles', roles)
+
+    def filter_vms(self):
+        cfg = config.vms
+        vms = [cfg.extend(i['vms']) for i in config.tenants if 'vms' in i]
+        opts = {'search_opts': {'all_tenants': 1}}
+        return [i for i in self.src_cloud.novaclient.servers.list(**opts)
+                if i.name in vms]
+
+    def filter_flavors(self):
+        flavors = [i['name'] for i in config.flavors]
+        return self.__get_nova_resources('flavors', flavors)
+
+    def filter_keypairs(self):
+        keypairs = [i['name'] for i in config.keypairs]
+        return self.__get_nova_resources('keypairs', keypairs)
+
+    def filter_security_groups(self):
+        sgs = [sg for i in config.tenants if 'security_groups' in i
+               for sg in i['security_groups']]
+        return self.__get_nova_resources('security_groups', sgs)
+
+    def filter_images(self):
+        images = [i['name'] for i in config.images]
+        return [i for i in self.src_cloud.glanceclient.images.list()
+                if i.name in images]
+
+    def filter_volumes(self):
+        volumes = [i['name'] for i in config.cinder_volumes]
+        opts = {'search_opts': {'all_tenants': 1}}
+        return [i for i in self.src_cloud.cinderclient.volumes.list(**opts)
+                if i.display_name in volumes]
+
+    def __get_neutron_resources(self, res, names):
```

Comment:
You know of the fact double underscores in the beginning of attribute name have [special meaning](https://docs.python.org/2/tutorial/classes.html#private-variables-and-class-local-references) in python, don't you? Make sure this is something you really need.